aptos-move/framework/aptos-framework/sources/governance_proposal.move

```diff
@@ -7,42 +7,33 @@ module aptos_framework::governance_proposal {
     use std::string::{String, length, utf8};
     use std::error;
 
-    const ECODE_LOCATION_TOO_LONG: u64 = 1;
-    const ETITLE_TOO_LONG: u64 = 2;
-    const EDESCRIPTION_TOO_LONG: u64 = 3;
+    const ETOO_LONG: u64 = 1;
 
     struct GovernanceProposal has store, drop {
-        /// The location (e.g. url) where the proposal resolution script's code can be accessed.
-        /// Maximum length allowed is 256 chars.
-        code_location: String,
-        /// Title of the proposal.
-        /// Maximum length allowed is 256 chars.
-        title: String,
-        /// Description of the proposal.
-        /// Maximum length allowed is 256 chars.
-        description: String,
+        // Location where metadata such as the proposal's execution script content, description, etc. are hosted.
+        metadata_location: String,
+        // The hash of the metadata to allow easy verification when a user votes that the metadata hosted at a url is
+        // correct.
+        metadata_hash: String,
```

Comment:
Do you want the code hash as well or only the metadata?

Reply:
I was thinking this can help people verify if the content at the provided url has been changed after the fact. The code can stay the same for example but description can change. I might be overthinking a bit though - this depends on how much validation we want to do in the CLI.

Reply:
Agree. I feel it makes sense for the Title and Description to be flexible but restrict the change of the code. If that the case can we make the hash a `code_hash` instead of `metadata_hash`? Also just brainstorming, maybe we can keep both code_location and metadata_location, so that one is unchangeable and the other is flexible.

Reply:
I think we should have a hash for the whole thing for now, allowing people to change data later is a nice to have.

Reply:
Yes what Daniel said. The code hash (execution_hash) is a separate field and meant for on-chain authorization when the vote is executed/resolved. The metadata hash is more meant for off chain usage to allow users to verify that the metadata such as description, in addition to the code, that they're looking at has not been maliciously altered.